grakel_replace/mixed_single_task_gp_usage_example.py

```diff
@@ -0,0 +1,102 @@
+import networkx as nx
+import torch
+from botorch.models.gp_regression_mixed import CategoricalKernel, ScaleKernel
+from gpytorch.distributions.multivariate_normal import MultivariateNormal
+from gpytorch.kernels import AdditiveKernel, MaternKernel
+from grakel_replace.mixed_single_task_gp import MixedSingleTaskGP
+from grakel_replace.torch_wl_kernel import TorchWLKernel
+
+TRAIN_CONFIGS = 10
+TEST_CONFIGS = 10
+TOTAL_CONFIGS = TRAIN_CONFIGS + TEST_CONFIGS
+
+N_NUMERICAL = 2
+N_CATEGORICAL = 2
+N_CATEGORICAL_VALUES_PER_CATEGORY = 3
+N_GRAPH = 2
+
+kernels = []
+
+# Create numerical and categorical features
+X = torch.empty(size=(TOTAL_CONFIGS, N_NUMERICAL + N_CATEGORICAL), dtype=torch.float64)
+if N_NUMERICAL > 0:
+    X[:, :N_NUMERICAL] = torch.rand(
+        size=(TOTAL_CONFIGS, N_NUMERICAL),
+        dtype=torch.float64,
+    )
+
+if N_CATEGORICAL > 0:
+    X[:, N_NUMERICAL:] = torch.randint(
+        0,
+        N_CATEGORICAL_VALUES_PER_CATEGORY,
+        size=(TOTAL_CONFIGS, N_CATEGORICAL),
+        dtype=torch.float64,
+    )
+
+# Create random graph architectures
+graphs = []
+for _ in range(TOTAL_CONFIGS):
+    G = nx.erdos_renyi_graph(n=5, p=0.5)  # Random graph with 5 nodes
+    graphs.append(G)
+
+# Create random target values
+y = torch.rand(size=(TOTAL_CONFIGS,), dtype=torch.float64)
+
+# Setup kernels for numerical and categorical features
+if N_NUMERICAL > 0:
+    matern = ScaleKernel(
+        MaternKernel(
+            nu=2.5,
+            ard_num_dims=N_NUMERICAL,
+            active_dims=tuple(range(N_NUMERICAL)),
+        ),
+    )
+    kernels.append(matern)
+
+if N_CATEGORICAL > 0:
+    hamming = ScaleKernel(
+        CategoricalKernel(
+            ard_num_dims=N_CATEGORICAL,
+            active_dims=tuple(range(N_NUMERICAL, N_NUMERICAL + N_CATEGORICAL)),
+        ),
+    )
+    kernels.append(hamming)
+
+# Combine numerical and categorical kernels
+combined_num_cat_kernel = AdditiveKernel(*kernels) if kernels else None
+
+# Create WL kernel for graphs
+wl_kernel = TorchWLKernel(n_iter=5, normalize=True)
```

Comment:
I'm guessing there's no way to really make it such that we could pass the `TorchWLKernel` to the `AdditiveKernel`, i.e. you would use it just like any other kernel type?

Reply:
We could define a `WLKernel` class that extends `gpytorch.kernels.Kernel` and use that class instead of `TorchWLKernel`.